pinot-spi/src/main/java/org/apache/pinot/spi/environmentprovider/PinotEnvironmentProvider.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.environmentprovider;
+
+import java.util.Map;
+import org.apache.pinot.spi.env.PinotConfiguration;
+
+/**
+ *  Environment Provider interface implemented by different cloud
+ *  providers to customize the base pinot configuration to add environment
+ *  variables & instance specific configuration
+ */
+public interface PinotEnvironmentProvider {
+
+  String INSTANCE_FAILURE_DOMAIN = "pinot.environment.instance.failureDomain";
+  /**
+   * Customize base pinot configuration to add environment variable & instance specific configuration
+   * @param baseConfiguration Base configuration provided by the invoking service
+   * @return Custom Pinot Configuration
+   */
+  PinotConfiguration getEnvironment(Map<String, Object> baseConfiguration);
```

Comment:
Why do we need to pass in the instance config twice (once in `init` and once `here`)? We should only pass it once.
Since the environment properties will be stored as a map field in the ZNRecord, we should probably just return a `Map<String, String>` here.
```suggestion
  Map<String, String> getEnvironment();
```

Reply:
Made the changes. Removed passing of instance config in two places.